File: max-consecutive-ones.py
class Solution(object):
    def findMaxConsecutiveOnes(self, nums):
        max_length = 0
        current_length = 0
        for i in nums:
            if i == 1:
                current_length += 1
            else:
                current_length = 0
            if max_length < current_length:
                max_length = current_length
        return max_length

File: max-consecutive-ones.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int current_ones = 0;
        int max_ones = 0;
        for (int x : nums) {
            if (x == 1) {
                current_ones += 1;
            } else {
                current_ones = 0;
            }
            
            if (current_ones > max_ones) {
                max_ones = current_ones;
            }
        }
        return max_ones;
    }
};
